ledger/src/blockstore_processor.rs

```diff
@@ -661,9 +661,9 @@ pub fn process_blockstore_from_root(
             .set_roots(std::iter::once(&start_slot))
             .expect("Couldn't set root slot on startup");
     } else {
-        assert!(
-            blockstore.is_root(start_slot),
-            "starting slot isn't root and can't update due to being secondary blockstore access: {}", start_slot
+        warn!(
+            "Starting slot {} isn't root and can't update due to being secondary blockstore access",
+            start_slot
```

Comment:
This function, `process_blockstore_from_root()`, later calls `load_frozen_forks()` which queries the blockstore for a root:
https://github.com/solana-labs/solana/blob/359e2de090607ded83c1d329ffcf65d2e797ffde/ledger/src/blockstore_processor.rs#L1127

However, the function also checks `bank_forks` and takes the max:
https://github.com/solana-labs/solana/blob/359e2de090607ded83c1d329ffcf65d2e797ffde/ledger/src/blockstore_processor.rs#L1128-L1129

So, I don't think it is actually a problem not to set `start_slot` as a root in the blockstore. As a quick test, I tried replaying a blockstore that has no roots:
```
Ledger has data for 103 slots 121519298 to 121519400
  with 0 rooted slots from 0 to 0
  and 103 slots past the last root
```
I got the same hashes as I did when running with current tip of master; and note this line of output which corresponds to the lines I called out above:
```
[2022-03-22T20:24:37.754080813Z INFO  solana_ledger::blockstore_processor]
load_frozen_forks() latest root from blockstore: 0, max_root: 121519300
```

Reply:
@carllin - I think Yueh-Hsuan and I should be able to handle rest of review between us, but from `git blame`, I'd appreciate any thoughts on this modification

Reply:
This functional change here is switching from an assert to a warning?

> So, I don't think it is actually a problem not to set start_slot as a root in the blockstore. As a quick test, I tried replaying a blockstore that has no roots:

`Blockstore::is_root` doesn't actually modify the database right?

Reply:
> This functional change here is switching from an assert to a warning?

Yes

> `Blockstore::is_root` doesn't actually modify the database right?

Also yes, `is_root()` only checks if the given slot is a root